libr/bin/format/pe/pemixed.c

```diff
@@ -0,0 +1,174 @@
+#include <stdio.h>
+#include "pemixed.h" 
+
+static bool check_il_only(ut32 flags);
+
+static int r_bin_pemixed_init(struct r_bin_pemixed_obj_t* bin, struct PE_(r_bin_pe_obj_t)* pe_bin){
+    struct PE_(r_bin_pe_obj_t)* sub_bin_dos;
+    struct PE_(r_bin_pe_obj_t)* sub_bin_native;
+    struct PE_(r_bin_pe_obj_t)* sub_bin_net;
+
+    sub_bin_dos = r_bin_pemixed_init_dos(pe_bin);
+    if(sub_bin_dos){
+        bin->sub_bin_dos = sub_bin_dos;   
+    }
+
+    sub_bin_native = r_bin_pemixed_init_native(pe_bin);
+    if(sub_bin_native){
+        bin->sub_bin_native = sub_bin_native;
+    }
+
+    sub_bin_net = pe_bin;
+    bin-> sub_bin_net = sub_bin_net;
+
+    return true;
+}
+
+//carves out dos from original pe
+//TODO: return mz file instead pe
+struct PE_(r_bin_pe_obj_t)* r_bin_pemixed_init_dos(struct PE_(r_bin_pe_obj_t)* pe_bin){
+    struct PE_(r_bin_pe_obj_t)* sub_bin_dos = R_NEW0(struct PE_(r_bin_pe_obj_t));
+    ut64 tmp_buf;
+
+    ut64 pe_hdr_off = pe_bin->dos_header->e_lfanew;
+
+    //idk if this is the most efficient way but could not find a function to read 
+    //RBuffer into another RBuffer
+    if(!(tmp_buf = malloc(pe_hdr_off))){
+        return NULL;
+    }
+    
+    if((r_buf_read_at(pe_bin->b, 0, tmp_buf, pe_hdr_off)) == -1){
+        printf ("Error reading to buffer\n");
+        return NULL;
+    }
+
+    sub_bin_dos->b = r_buf_new ();
+    sub_bin_dos->size = pe_hdr_off;
+
+    if(!r_buf_set_bytes (sub_bin_dos->b, tmp_buf, pe_hdr_off)){
+        PE_(r_bin_pe_free)(sub_bin_dos);
+    }
+ 
+    sub_bin_dos->dos_header = pe_bin->dos_header;
+     
+    free(tmp_buf);
```

Comment:
wtf , tmp_buf is ut64 ?

Reply:
lol well i figured since malloc returned a 64bit mem address so... 
Changed it to char* instead